src/utils/gitUtils.ts

```diff
@@ -0,0 +1,145 @@
+    try {
+        await repo.commit(commitMsg, commitOptions);
+    } catch (err) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (!/nothing to commit/.test(err.stdout)) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+            console.debug(parseError(err));
+            // ignore empty commit errors which will happen if a user initializes a blank folder or have changes in a nested git repo
+            throw (err);
+        }
+    }
+}
+
+export async function promptForDefaultBranch(context: IActionContext, repo: Repository): Promise<void> {
+    const defaultBranch: string | undefined = await tryGetDefaultBranch(repo)
+    context.telemetry.properties.defaultBranch = defaultBranch;
+
+    if (defaultBranch && repo.state.HEAD?.name !== defaultBranch) {
+        context.telemetry.properties.cancelStep = 'defaultBranch';
+        context.telemetry.properties.notOnDefault = 'true';
+
+        const checkoutButton: MessageItem = { title: localize('checkout', 'Checkout "{0}"', defaultBranch) };
+        const result: MessageItem = await ext.ui.showWarningMessage(localize('deployBranch', 'It is recommended to connect your SWA to the default branch "{0}".  Would you like to continue with branch "{1}"?',
+            defaultBranch, repo.state.HEAD?.name), { modal: true }, checkoutButton, { title: localize('continue', 'Continue') });
+        if (result === checkoutButton) {
+            await repo.checkout(defaultBranch);
+            context.telemetry.properties.checkoutDefault = 'true';
+        } else {
+            context.telemetry.properties.checkoutDefault = 'false';
+        }
+    }
+}
+
+export async function tryGetDefaultBranch(repo: Repository): Promise<string | undefined> {
+    // currently git still uses master as the default branch but will be updated to main so handle both cases
+    // https://about.gitlab.com/blog/2021/03/10/new-git-default-branch-name/#:~:text=Every%20Git%20repository%20has%20an,Bitkeeper%2C%20a%20predecessor%20to%20Git.
+    const defaultBranches: string[] = ['main', 'master'];
+    try {
+        defaultBranches.unshift(await repo.getConfig('init.defaultBranch'));
+    } catch (err) {
+        // if no local config setting is found, try global
+        try {
+            defaultBranches.push(await repo.getGlobalConfig('init.defaultBranch'));
+        } catch (err) {
+            // VS Code's git API doesn't fail gracefully if no config is found, so swallow the error
+        }
+    }
+
+    const localBranches: Ref[] = await repo.getBranches({ remote: false });
+    // order matters here because we want the setting, main, then master respectively so use indexing
+    for (let i = 0; i < localBranches.length; i++) {
+        if (localBranches.some(lBranch => lBranch.name === defaultBranches[i])) {
```

Comment:
Since `localBranches` and `defaultBranches` aren't guaranteed to be the same length I don't think it's safe to use `i` to index into both

Reply:
Good catch.  Meant to use defaultBranches.length as the for loop condition.